pkg/resources/service/pingsource_test.go

```diff
@@ -0,0 +1,93 @@
+package service
+
+import (
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/triggermesh/tm/pkg/client"
+	"github.com/triggermesh/tm/pkg/file"
+)
+
+func TestPingSource(t *testing.T) {
+	namespace := "test-namespace"
+	if ns, ok := os.LookupEnv("NAMESPACE"); ok {
+		namespace = ns
+	}
+
+	testCases := []struct {
+		name              string
+		service           *Service
+		pingSourceCreated bool
+		reason            string
+	}{
+		{
+			name: "valid pingsource",
+			service: &Service{
+				Name:   "valid-pingsource",
+				Source: "gcr.io/google-samples/hello-app:1.0",
+				Schedule: []file.Schedule{
+					{
+						Cron:     "*/1 * * * *",
+						JSONData: `{"some":"data"}`,
+					},
+				},
+			},
+			pingSourceCreated: true,
+		}, {
+			name: "malformed schedule",
+			service: &Service{
+				Name:   "malformed-schedule",
+				Source: "gcr.io/google-samples/hello-app:1.0",
+				Schedule: []file.Schedule{
+					{
+						Cron: "not a schedule",
+					},
+				},
+			},
+			pingSourceCreated: false,
+		},
+	}
+
+	client.Dry = false
+	client.Wait = true
+	serviceClient, err := client.NewClient(client.ConfigPath(""))
```

Comment:
Do I understand correctly that those tests require a real cluster? If yes, shall we maybe move them under a `integration` sub-directory somewhere to make it clear they depend on third-party systems?

Reply:
all correct - some of our unit tests went through a very tough path and became integration tests. Overall, our tests require a lot of love and I'd dedicate separate issue(s) for such tasks :slightly_smiling_face:

Reply:
https://github.com/triggermesh/tm/issues/198